### PR TITLE
feat: [PLG-3076]: agent simplification flow

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by satyam-harness — DISMISSED** [review]

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx [hunk 1]
```diff
@@ -67,13 +166,58 @@ export default function CreateGitopsAgentModal({
...
+          width={isVerifiedOnce ? 200 : 120}
+        />
+      ) : null}
+    </>
+
+    // <Button
+    //   width={220}
+    //   text={getString(
+    //     agentInfo?.identifier ? 'cd.getStartedWithCD.updateGitopsAgent' : 'cd.getStartedWithCD.createGitopsAgent'
+    //   )}
```
Comment (vikrant-harness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx:218]: commented code?

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/CDOnboardingWizardWithCLI.module.scss [hunk 2]
```diff
@@ -65,6 +65,26 @@
...
 }
 
+.downloadButton {
+  height: 28px;
+  margin-bottom: 12px;
+  max-width: 180px;
+}
+
+.subHeading {
+  padding-bottom: 12px !important;
```
Comment (vikrant-harness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/CDOnboardingWizardWithCLI.module.scss:75]: can we use css variables here?

**Review by satyam-harness — DISMISSED** [review]

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx [hunk 3]
```diff
@@ -31,14 +53,91 @@ export default function CreateGitopsAgentModal({
...
+  const linkRef = React.useRef<HTMLAnchorElement>(null)
+  const { showError } = useToaster()
+
+  const { mutate: createAgent } = useAgentServiceForServerCreate({})
+  const { projectIdentifier, orgIdentifier, accountId } = useParams<ProjectPathProps>()
+  const handleCreateOrEdit = async (payload: V1Agent): Promise<any> => {
+    const response = await createAgent(payload)
+    return {
+      status: response
+      // nextCallback: afterSuccessHandler.bind(null, response)
```
Comment (vikrant-harness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx:72]: commented code ?

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx [hunk 4]
```diff
@@ -31,14 +53,91 @@ export default function CreateGitopsAgentModal({
...
+  }
+  const { refetch: refetchHelmChart, data } = useAgentServiceForServerGetDeployHelmChart({
+    queryParams: defaultQueryParams,
+    agentIdentifier: 'harnessagent' as string,
+    lazy: true
+  })
+
+  const handleDownload = (): void => {
+    if (linkRef?.current) {
+      // This code is taken from Chat GPT
```
Comment (vikrant-harness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/DeploymentFlowTypes/CreateAgentModal.tsx:122]: 😅

**Review by satyam-harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Pull request merged**